src/geo/projection/globe_util.js

```diff
@@ -75,6 +122,127 @@ export function globeTileBounds(id: CanonicalTileID): Aabb {
     return new Aabb(bMin, bMax);
 }
 
+export function aabbForTileOnGlobe(tr: Transform, numTiles: number, tileId: CanonicalTileID): Aabb {
+    const scale = numTiles / tr.worldSize;
+    const to1to1Units = (cornerMin: Vec3, cornerMax: Vec3) => {
+        vec3.scale(cornerMin, cornerMin, scale);
+        vec3.scale(cornerMax, cornerMax, scale);
+    };
+
+    const mx = Number.MAX_VALUE;
+    const cornerMax = [-mx, -mx, -mx];
+    const cornerMin = [mx, mx, mx];
+    const m = calculateGlobeMatrix(tr);
```

Comment:
We added globe matrix caching in #11499 — let's try making use of it here too to avoid redundant calculations / allocations.

Reply:
Ha! That's a very good point and I am surprised back then when I was resolving conflicts I didn't catch this. Thank you @mourner  I will address it in a separate PR.